plugin.video.fanfilm/resources/lib/sources/pl/alltube.py

```diff
@@ -76,9 +76,14 @@ def search(self, title, localtitle, year, search_type):
             titles= []
             titles.append(cleantitle.normalize(cleantitle.getsearch(title)))
             titles.append(cleantitle.normalize(cleantitle.getsearch(localtitle)))
-            
+			
             for title in titles:
-                r = client.request(urlparse.urljoin(self.base_link, self.search_link), post={'search': cleantitle.query(title)})
+                headers = {
+                'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/68.0.3435.0 Safari/537.36',
```

Comment:
Uważam, że `User-Agent` tutaj nie powinien być definiowany. Przez to jest copy-paste między wtyczkami.
To musi być we wspólnym miejscu. I mam wrażenie, że gdzieś jest.
Jeszcze raz wezwę na pomoc @xulek, @notoco, @CherryKodi, @podpis...

Reply:
Znalazłem w regex.py definicję getUrl(), który obsługuje ciasteczka, nagłówki i BasicAuth, jest tam wpisany z palca User-Agent, dodawany do każdego zapytania. Żeby było śmiesznie, nigdzie nie znalazłem wykorzystania tego. Staram się ożywić resztę źródeł z /pl, więc jak mi się uda, to przetestuję.

Ponadto w cfscrape.py jest, zdaje się, cała obsługa Cloud Flare, włącznie z obsługą odpowiedzi 503, losowym wybieraniem User-Agent i innymi smaczkami. Nie testowałem, z opisu wnioskuję, że wymaga Node.js na maszynie.

Reply:
getUrl w regexp. Hmmm.. ciekawa koncepcja :-)
Poważniej, to w całości brzmi naprawdę ciekawie. Dobrze,że znalazłeś, bo ja mam gdzieś na kolanie napisaną obsługę (jak  żonie przestało coś działać) i pewnie bym wrzucał. A jak jest to lepiej użyć / poprawić istniejące.

Generalnie to większość jest do przepisania. Pewne prace się rozpoczęły i utknęły z braku czasu. Rozpoczynamy os PTW, które ma być jądrem biblioteki. 
Ja jeszcze mam w zamysłach dalekich system generycznego dodatku, który by odwalał większość typowej roboty. Tak, żeby wtyczka to była sama logika dotycząca obsługiwanej strony.
Trochę OT wyszedł.